rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -1676,6 +1683,96 @@ func Qualify_Commodity_Forward: <"Qualifies a product as a Forward that will be
             [synonym ISDA_Taxonomy_v2 value "Commodity_Forward"]
     set is_product:
         Qualify_AssetClass_Commodity(economicTerms) = True
-            and ((economicTerms -> payout -> settlementPayout, economicTerms -> payout -> fixedPricePayout) only exists
-                or // Fixed Price Forward
-                (economicTerms -> payout -> settlementPayout, economicTerms -> payout -> commodityPayout) only exists) // Floating Price Forward
+            and economicTerms -> payout count = 2 
+            and ((economicTerms -> payout -> SettlementPayout exists and economicTerms -> payout -> FixedPricePayout exists) // Fixed Price Forward
+                or (economicTerms -> payout -> SettlementPayout exists and economicTerms -> payout -> CommodityPayout exists)) // Floating Price Forward
+
+func SingleAssetPayoutOnlyExists: <"Exactly one AssetPayout exists and no other payouts exist.">
```

Comment:
Remove all the "SinglePayoutOnlyExists" functions, keep the logic in-line instead as it's quite simple.

Reply:
Done